src/core/QCheck.mli

```diff
@@ -190,6 +190,9 @@ module Gen : sig
   val shuffle_l : 'a list -> 'a list t
   (** Creates a generator of shuffled lists. *)
 
+  val shuffle_w_l : (int * 'a) list -> 'a list t
+  (** Creates a generator of weighted shuffled lists. *)
```

Comment:
this needs a `@since` annotation and, I'm afraid, a lot more explanations :)
An example or two would be very useful (possibly also the intended use case).

Reply:
Sure, that's a great idea! 
Had the same feeling.